app/charts/column/columns-stacked-state.tsx

```diff
@@ -257,9 +260,21 @@ const useColumnsStackedState = (
       dimensionFilter: segmentFilter,
     });
 
-    return orderBy(uniqueSegments, sorters, getSortingOrders(sorters, sorting));
+    const allSegments = orderBy(
+      allUniqueSegments,
+      sorters,
+      getSortingOrders(sorters, sorting)
+    );
+
+    return {
+      segments: allSegments,
+      plottableSegments: allSegments.filter((d) =>
+        allPlottableSegments.includes(d)
+      ),
+    };
```

Comment:
I am under the impression that the `segments` returned here is only used for the order of the tooltip values. If so, I think the order could only rely on plottableSegments, and this memo could only return plottableSegments.

Reply:
I also thought that this is the case, but we also use them in color domain, to display the legend – if we just return `plottableSegments`, we would remove a color legend item permanently once de-selected (and completely remove the chart is we de-select them all 😅)

Reply:
Ahah, understood 👍